include/base64.h

```diff
@@ -9,88 +9,166 @@
 #ifndef _SQUID_BASE64_H
 #define _SQUID_BASE64_H
 
-#if HAVE_NETTLE_BASE64_H && HAVE_NETTLE30_BASE64
+#if HAVE_NETTLE_BASE64_H && HAVE_NETTLE34_BASE64
 #include <nettle/base64.h>
 
-#else /* Base64 functions copied from Nettle 3.0 under GPLv2, with adjustments */
+#else /* Base64 functions copied from Nettle 3.4 under GPLv2, with adjustments */
 
-#ifdef __cplusplus
-extern "C" {
-#endif
+/* base64.h
 
-// Decoding functions
+   Base-64 encoding and decoding.
 
-/// Maximum length of output for base64_decode_update.
-/// We have at most 6 buffered bits, and a total of (length + 1) * 6 bits.
-#   define BASE64_DECODE_LENGTH(length) ((((length) + 1) * 6) / 8)
+   Copyright (C) 2002 Niels Möller, Dan Egnor
 
-struct base64_decode_ctx
-{
-    unsigned word;   /* Leftover bits */
-    unsigned bits;   /* Number buffered bits */
+   This file is part of GNU Nettle.
 
-    /* Number of padding characters encountered */
-    unsigned padding;
-};
+   GNU Nettle is free software: you can redistribute it and/or
+   modify it under the terms of either:
 
-void base64_decode_init(struct base64_decode_ctx *ctx);
+     * the GNU Lesser General Public License as published by the Free
+       Software Foundation; either version 3 of the License, or (at your
+       option) any later version.
 
-/* Returns 1 on success, 0 on error. DST should point to an area of
- * size at least BASE64_DECODE_LENGTH(length). The amount of data
- * generated is returned in *DST_LENGTH.
- */
-int base64_decode_update(struct base64_decode_ctx *ctx,
-                         size_t *dst_length,
-                         uint8_t *dst,
-                         size_t src_length,
-                         const uint8_t *src);
+   or
 
-/* Returns 1 on success. */
-int base64_decode_final(struct base64_decode_ctx *ctx);
+     * the GNU General Public License as published by the Free
+       Software Foundation; either version 2 of the License, or (at your
+       option) any later version.
+
+   or both in parallel, as here.
+
+   GNU Nettle is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   General Public License for more details.
+
+   You should have received copies of the GNU General Public License and
+   the GNU Lesser General Public License along with this program.  If
+   not, see http://www.gnu.org/licenses/.
+*/
+
+#ifdef __cplusplus
+extern "C" {
+#endif
 
-// Encoding functions
+//#define BASE64_BINARY_BLOCK_SIZE 3
+//#define BASE64_TEXT_BLOCK_SIZE 4
```

Comment:
I suggest removing these two (presumably unused) #defines instead of commenting them out.